cpp/include/Ice/Proxy.h

```diff
@@ -1107,22 +1106,19 @@ class ICE_API ObjectPrx : public ::std::enable_shared_from_this<ObjectPrx>
         return [outAsync]() { outAsync->cancel(); };
     }
 
-    virtual ::std::shared_ptr<ObjectPrx> _newInstance() const;
-    ObjectPrx() = default;
-    friend ::std::shared_ptr<ObjectPrx> IceInternal::createProxy<ObjectPrx>();
     /// \endcond
 
 private:
 
-    void setup(const ::IceInternal::ReferencePtr&);
-    friend class ::IceInternal::ProxyFactory;
-
-    ::IceInternal::ReferencePtr _reference;
+    const ::IceInternal::ReferencePtr _reference;
```

Comment:
ObjectPrx holding this const reference is the pimpl technique:
https://en.cppreference.com/w/cpp/language/pimpl

Note:
 - it's pretty C++ specific so it's odd to use this technique in other languages
 - the "pointer to implementation" is typically a unique_ptr, not a shared_ptr. We should consider switching to unique_ptr when we refactor IceInternal::Reference